src/agrep.c

```diff
@@ -652,6 +666,10 @@ Copyright (c) 2001-2009 Ville Laurikari <vl@iki.fi>.\n"));
   if (show_help)
     tre_agrep_usage(0);
 
+  /* Fail without reading if count number is 0 */
+  if (max_count == 0)
+    return EXIT_FAILURE;
```

Comment:
We don't use `EXIT_SUCCESS` or `EXIT_FAILURE` elsewhere, so we shouldn't use them here.

Reply:
Then would `return 1;` be more appropriate?